projects/pastanaga-angular/src/lib/controls/textfield/select/select.component.ts

```diff
@@ -0,0 +1,373 @@
+                        // then hold delete. Toggling to `1, 1` and then back to
+                        // `0, 0` seems to completely fix it.
+                        element.setSelectionRange(1, 1);
+                        element.setSelectionRange(0, 0);
+                    }
+                });
+            });
+        }
+    }
+
+    findValueByLabel(): string | null{
+        if (!!this.ngContentOptions && this.ngContentOptions.length > 0) {
+            const selectedComponent = this.ngContentOptions.find((optionComponent) => optionComponent.text === this.displayedValue);
+            return selectedComponent ? selectedComponent.value : null;
+        } else {
+            const selectedModel = this._options.find((option: OptionModel) => option.label === this.displayedValue) as OptionModel;
+            return selectedModel ? selectedModel.value : null;
+        }
+    }
+
+    findLabelByValue(value?: string): string | undefined {
+        let label: string | undefined;
+        if (!!this.ngContentOptions && this.ngContentOptions.length > 0) {
+            const selectedOption = this.ngContentOptions.find((option) => option.value === this.model);
+            label = !!selectedOption ? selectedOption.text : undefined;
+        } else {
+            const selectedOption =  this._options.find((option: OptionModel) => option.value === value);
+            label = !!selectedOption ? (selectedOption as OptionModel).label : undefined;
+        }
+        if (!label && this.suggestionMode) {
+            label = value;
+        }
+        return label;
+    }
+
+    dropDownClosed() {
+        this.optionsClosed.next();
+        this.expanded.emit(false);
+    }
+
+    dropDownOpened() {
+        if (!this._optionsAreNgContent) {
+            setTimeout(() => {
+                this.ngContentOptions?.forEach((option: OptionComponent, index) => {
+                    option.selectOption.pipe(takeUntil(this.optionsClosed)).subscribe(() => {
+                        this.onValueChange(option.value);
+                    });
+                });
+            });
+        }
+        this.expanded.emit(true);
+    }
+
+    /**
+     * browse options and update selected status
+     */
+    private flagSelectedOption() {
+        if (!!this.ngContentOptions) {
+            this.ngContentOptions.forEach((option) => {
+                const isSelected = option.value === this.model;
+                if (option._selected !== isSelected) {
+                    option._selected = isSelected;
+                    option.refresh();
+                }
+            });
+        }
+        if (!!this._options) {
+            this._options.filter(opt => opt.type === ControlType.option).forEach((option: OptionModel) => {
+                const isSelected = option.value === this.model;
+                if (option.selected !== isSelected) {
+                    option.selected = isSelected;
+                }
+            });
+        }
+    }
```

Comment:
I feel like you might trigger a lot of change detection cycles by doing so. You might have no other choice for  `ngContentOptions `, but for `_options` it is probably better to do all the changes in a loop in a variable, and only after that affecting the variable to `this._options`

Reply:
Sorry I don't get it... I'm not changing the reference to this_options...

Reply:
you are changing option's `selected` value in a forEach loop, meaning change detection might be triggered for each option changed and we don't want that.
What we want is to have only one change detection triggered after the loop is done, so something like that:
```typescript
        if (!!this._options) {
            this._options = this._options.map((option: OptionModel) => {
                if (option.type === ControlType.option) {
                    const isSelected = option.value === this.model;
                    if (option.selected !== isSelected) {
                        option.selected = isSelected;
                    }
                }
                return option;
            });
        }
```